### DISCUSSION TITLE
Access original coordinate after stacking?

### DISCUSSION THREAD

**Discussion by ghbrown** [pydata/xarray#9131] [Q&A] [answered]
What is the best way to access the original values of stacked coordinates?
I can manage this via unstacking, but ideally (for my use case) there is another way.

```python
import numpy as np
import xarray as xr

n = 40
lat  = np.ones(n)
lon  = np.arange(n)
time = np.ones(n)
A  = np.zeros((n,n,n))
da = xr.DataArray(data=A,
        dims=['lat','lon','time'],
        coords=dict(lat=lat,lon=lon,time=time))
da = da.stack(latlon=('lat','lon'))

print(f'neither of these are the desired length ({n}):')
print(da.lon.shape)
print(da.coords['lon'].shape)
```
Edit: switched one of the stacked coordinates to `arange` for clarity

**Comment by dcherian** [comment, marked as answer]
You can use the underlying pandas multiindex: `da.xindexes["latlon"].index` The `levels` and `names` properties will be useful: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.MultiIndex.html

It'll be easier to interpret if you use `lon = np.arange(n)`

**Reply by ghbrown** [reply]
Exactly what I was looking for, thanks!

Here's example code to extract `lat` for future readers' reference:
```python
# recover array corresponding to lat
multi = da.xindexes['latlon'].index  # pandas.MutiIndex for latlon
ind_lat = multi.names.index('lat')   # get index of lat using list index() method
lat_index = multi.levels[ind_lat]    # index into levels
print(f'pandas.Index for lat: {lat_index}')
print(f'array for lat: {lat_index.values}')
```
